lib/WTSI/NPG/HTS/PacBio/Annotator.pm

```diff
@@ -93,45 +134,58 @@ sub make_sample_metadata {
                      cohort           => $SAMPLE_COHORT,
                      donor_id         => $SAMPLE_DONOR_ID};
 
-  return $self->_make_single_value_metadata($sample, $method_attr);
+  return $self->_make_multi_value_metadata(\@samples, $method_attr);
 }
 
+=head2 make_library_metadata
+
+  Arg [n]      PacBio run records,
+               Array[WTSI::DNAP::Warehouse::Schema::Result::PacBioRun].
+
+  Example    : my @avus = $ann->make_library_metadata(@run_records);
+  Description: Return secondary AVU metadata for a run.
+  Returntype : Array[HashRef]
+
+=cut
+
 sub make_library_metadata {
   my ($self, @run_records) = @_;
 
-  my @avus;
-  my @library_ids =
-    uniq map { $_->pac_bio_library_tube_legacy_id } @run_records;
-
-  foreach my $library_id (@library_ids) {
-    push @avus, $self->make_avu($LIBRARY_ID, $library_id);
-  }
+  my $method_attr = {pac_bio_library_tube_legacy_id => $LIBRARY_ID};
+  my @avus = $self->_make_multi_value_metadata(\@run_records, $method_attr);
 
-  my $num_libraries = scalar @library_ids;
+  my $num_libraries = scalar @run_records;
   if ($num_libraries > 1) {
-    push @avus, 'multiplex', q[1];
-    push @avus, 'library_id_composite', (join q[;], @library_ids);
+    push @avus, 'multiplex', 1;
```

Comment:
Maybe;  push @avus, $self->make_avu('multiplex', 1);   ??
getting error:  FATAL WTSI.NPG.HTS.DataObject - The avu argument must be a HashRef